landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Calculate dimensionless discharge of stream sections based on Tang et
+al. (2019)
+"""
+
+import numpy as np
+
+from landlab import Component
+
+
+class DimensionlessDischarge(Component):
+    r"""Component that calculates dimensionless discharge of stream
+     segments.
+
+    The dimensionless discharge model calculates the unitless  discharge
+    value of streams and can be used to help determine locations of
+    debris flows. It uses an equation from Tang et al. (2019) to
+    calculate the dimensionless discharge as well as the threshold for
+    whether a debris flow will occur for a specified location.
+
+    .. math::
+
+        q* = \frac{q}{\sqrt{\frac{\rho_s-\rho}{\rho}*g*D_{50}^3}}
+
+     where :math:`q*` is the dimensionless discharge value for a stream
+     segment, :math:`q` is flux in a stream segment,  :math:`\rho_s`
+     is soil density, :math:`\rho` is water density, :math:`g` is the
+     gravitational constant, and :math:`D_50` is the average sediment
+     partical size in the stream segment area.
+
+     Constants C and N are coefficients used in the slope-dependent
+     equation
+
+     .. math::
+
+        q*_{thresold} = \frac{C}{(tan(\theta))^N}
+
+     to determine whether the dimensionless discharge calculated
+     exceeds thresholds for a sediment-laden (Upper Limit) or
+     water-producing only (Lower Limit) debris flow.  C and N are
+     empirically-derived by comparing dimensionless discharge estimates
+     against previous debris flow events.  The values will vary based on
+     local geology and soil type. In southern California values of C=12
+     and N=0.85 (upper limits, Tang et al, 2019) and C=4.29, N=0.78
+     (lower limits, Tang et al, 2019) have been used, while values of
+     C=0.195 and N=1.27 have been in used in the Italian Dolomites
+     (Gregoretti and Fontana, 2008). Default values are C=12 and N=0.85.
```

Comment:
This is a really nice overview of the component. Nice job!

Reply:
Thanks!

Reply:
Thanks!